e2e-tests/batch-mode-tests.sh

```diff
@@ -1,119 +1,160 @@
+  echo "${print_prefix} Total encounters synced to parquet ---> ${total_encounters_streamed}"
+  total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Observation/ | awk '{print $3}')
+  echo "${print_prefix} Total obs synced to parquet ---> ${total_obs_streamed}"
+  if [[ ${total_patients_streamed} == ${total_test_patients} && ${total_encounters_streamed} == ${total_test_encounters} && $total_obs_streamed == ${total_test_obs} ]] \
+    ; then
+    echo "${print_prefix} BATCH MODE WITH PARQUET SYNC EXECUTED SUCCESSFULLY USING $mode MODE"
+    cd "${HOME_PATH}"
+  else
+    echo "${print_prefix} BATCH MODE WITH PARQUET SYNC TEST FAILED USING $mode MODE"
+    exit 1
+  fi
+}
+
+function test_fhir_sink() {
+  local test_dir=$2
+  local mode=$3
+
+  echo "${print_prefix} $1 "
+  $1
+
+  cd "${test_dir}"
+  mkdir omrs_fhir_sink
+  echo "${print_prefix} Finding number of patients, encounters and obs in openmrs server"
+  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
+    http://localhost:8099/openmrs/ws/fhir2/R4/Patient/ 2>/dev/null >>./omrs_fhir_sink/patients.json
+  total_test_patients=$(jq '.total' ./omrs_fhir_sink/patients.json)
+
+  mkdir fhir
+  echo "${print_prefix} Counting number of patients, encounters and obs sinked to fhir files"
+  curl -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 \
+    http://localhost:8098/fhir/Patient/?_summary=count 2>/dev/null >>./fhir/patients.json
+  total_patients_sinked_fhir=$(jq '.total' ./fhir/patients.json)
+  echo "${print_prefix} Total patients sinked to fhir ---> ${total_patients_sinked_fhir}"
+  if [[ ${total_patients_sinked_fhir} == ${total_test_patients} ]]; then
+    echo "${print_prefix} BATCH MODE WITH FHIR SERVER SINK EXECUTED SUCCESSFULLY USING $mode MODE"
+    cd "${HOME_PATH}"
+  else
+    echo "${print_prefix} BATCH MODE WITH FHIR SERVER SINK TEST FAILED USING $mode MODE"
+    exit 1
+  fi
+}
+
+setup
+
+echo "${print_prefix} ---- STARTING PARQUET SINK TEST ----"
+
+
+test_parquet_sink "mvn exec:java -pl batch \"-Dexec.args=--openmrsServerUrl=http://localhost:8099/openmrs \
+        --openmrsUserName=admin --openmrsPassword=Admin123
+        --searchList=Patient,Encounter,Observation --batchSize=20 \
+        --targetParallelism=20 --fileParquetPath='${test_dir_fhir}'/" "${test_dir_fhir}" "FHIR_SEARCH"
+
+test_parquet_sink "mvn exec:java -pl batch \"-Dexec.args=--batchSize=20 --targetParallelism=20 \
+   --searchList=Patient,Encounter,Observation --fileParquetPath='${test_dir_jdbc}'/ \
+   --jdbcModeEnabled=true --jdbcUrl=jdbc:mysql://localhost:3306/openmrs \
+   --dbUser=mysqluser --dbPassword=mysqlpw --jdbcMaxPoolSize=50 \
+   --jdbcDriverClass=com.mysql.cj.jdbc.Driver" "${test_dir_jdbc}" "JDBC"
+
+echo "${print_prefix} ---- STARTING FHIR SINK TEST ----"
+
+test_fhir_sink "mvn exec:java -pl batch \"-Dexec.args=--searchList=Patient --batchSize=20 --targetParallelism=20 \
```

Comment:
I should have mentioned this in the origin e2e PR, but it is not too late :-)
I think you can merge the two `exec` commands for Parquet file generation and sending to HAPI FHIR sink together. IOW, in the same shot, you provide both `--fileParquetPath` and `--fhirSinkPath`. That should save us some time in e2e runs.

Reply:
We can do this only if we set `--searchList=Patient`. If we set `--searchList=Patient,Encounter,Observation` and run a FHIR sink pipeline:

```
mvn exec:java -pl batch "-Dexec.args=--openmrsServerUrl=http://localhost:8099/openmrs     --openmrsUserName=admin --openmrsPassword=Admin123     --searchList=Patient,Encounter,Observation --batchSize=20      --fhirSinkPath=http://localhost:8098/fhir --sinkUserName=hapi --sinkPassword=hapi "
```

It will fail with:
```
Caused by: ca.uhn.fhir.rest.server.exceptions.InvalidRequestException: HTTP 400 : Resource Encounter/1edbb772-53e2-4d29-9428-54fb8c217e74 not found, specified in path: Encounter.partOf
```

This seems to happen here:
https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/master/common/src/main/java/org/openmrs/analytics/FhirStoreUtil.java#L124-L135

Can we only sink `Patient` into the FHIR DB?

Reply:
Oh, I see, this is the referential integrity problem. @mozzy11 did you look into how to disable that for HAPI server?
@omarismail94 let's just add a TODO for now to merge them in the future.